core/src/main/java/nu/yona/server/rest/PassThroughHeadersHolder.java

```diff
@@ -7,16 +7,26 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import org.springframework.core.NamedThreadLocal;
 import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
 
 /**
  * Holds the headers that are received in an incoming HTTP request, which needs to be passed-through in outgoing HTTP requests.
  * See {@link HeadersServerInterceptor} for the place where the incoming headers are stored in the holder and
  * {@link HeadersClientInterceptor} for the place where the headers are read.
  */
+@Component
 public class PassThroughHeadersHolder
 {
-	private final Map<String, String> storedHeaders = new HashMap<>();
+	private final ThreadLocal<Map<String, String>> threadLocal = new NamedThreadLocal<Map<String, String>>(
```

Comment:
One thing I have experienced before with thread local is that for example IIS uses a thread pool and sometimes reuses a thread. Can we get the same issue with Apache? So should we not at least clear the holder on the `HeadersServerInterceptor` to be sure?

Reply:
Valid point. The threads can certainly be reused.

Reply:
I'm going to clear it as part of ``readFrom`` and ``importFrom``.